# 1. International debt data by the World Bank
Federal governments use debts to leverage the economy. Borrowed funds are typically used to invest in infrastructure, healthcare, education, etc. With recent developments loans are used to alleviate poverty and social inequality. World Bank is one of the major loan providers across the globe with affordable interest rates.

In this notebook we will perform a detailed analysis of debt data provided by the World Bank to answer the following questions:
- What is the total amount of debt and average debt owed in every category by countries?
- What are the top indebted countries?
- What are the most common debt types across the nations?

Let's inspect the dataset to get a sense of the data.

In [2]:
%%sql 
SELECT 
    * 
FROM 
    international_debt
FETCH FIRST 10 ROW ONLY

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
10 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', 72894453.7),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', 53239440.1),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', 61739336.9),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', 49114729.4),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', 39903620.1),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', 39107845.0),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', 23779724.3),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', 13335820.0),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', 100847181.9),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', 72894453.7)]

# 2. Distinct countries
Afghanistan owes some amount of debt in multiple categories, therefore countries are most likely to get the loans in multiple indicators.

Let's find out the number of unique countries in the dataset to proceed further with analysis.

In [3]:
%%sql 
SELECT 
    COUNT(DISTINCT country_name) AS distinct_countries
FROM 
    international_debt

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
1 rows affected.


[(124,)]

# 3. Distinct debt indicators
There are 124 distinct countries in the dataset. The indicator name column contains an explanation of debt and indicator code in a unique identifier for debt purpose. Finding unique indicator codes in the dataset can help us understand the possible reasons why particular countries take loans.

In [4]:
%%sql
SELECT 
    DISTINCT indicator_code AS distinct_debt_indicators
FROM 
    international_debt
ORDER BY 
    distinct_debt_indicators

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
25 rows affected.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',),
 ('DT.AMT.PBND.CD',),
 ('DT.AMT.PCBK.CD',),
 ('DT.AMT.PROP.CD',),
 ('DT.AMT.PRVT.CD',),
 ('DT.DIS.BLAT.CD',),
 ('DT.DIS.DLXF.CD',),
 ('DT.DIS.MLAT.CD',),
 ('DT.DIS.OFFT.CD',),
 ('DT.DIS.PCBK.CD',),
 ('DT.DIS.PROP.CD',),
 ('DT.DIS.PRVT.CD',),
 ('DT.INT.BLAT.CD',),
 ('DT.INT.DLXF.CD',),
 ('DT.INT.DPNG.CD',),
 ('DT.INT.MLAT.CD',),
 ('DT.INT.OFFT.CD',),
 ('DT.INT.PBND.CD',),
 ('DT.INT.PCBK.CD',),
 ('DT.INT.PROP.CD',),
 ('DT.INT.PRVT.CD',)]

# 4. The total amount of debt owed by countries around the globe
Totalling the amount of debt can help us to have a sense of the debt amount on a global scale. Let's find out the total debt amount in (bil, USD) owed worldwide.

In [5]:
%%sql
SELECT 
    ROUND(SUM(debt) / 100000000) AS total_debt
FROM 
    international_debt

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
1 rows affected.


[(30797.0,)]

# 5. The most indebted countries
All the countries require a different size of loan due to their current economic need and present economic state. We're going, to sum up the debt across all the debt indicators to find out what country was the most hungry for the World Bank's funds.

In [6]:
%%sql
SELECT 
    country_name, 
    SUM(debt) AS aggregated_debt
FROM 
    international_debt
GROUP BY 
    country_name
ORDER BY 
    aggregated_debt DESC
FETCH FIRST 1 ROW ONLY

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
1 rows affected.


[('China', 285793494734.2)]

# 6. Average debt across indicators
China isn't leading the list surprisingly. The 21st century is going to be a boom in the economic development of the Asian region. China used to be opposed and very limited to foreign equity investments until 1978 when the 'Open Door Policy' took place and liberated Chinese law to open up for FDA. Therefore, debt was the only major option for financing economic needs. A detailed breakdown of debt can be found [here](https://datatopics.worldbank.org/debt/ids/country/CHN).

Now let's take a look at the most common debt indicators by the average amount of debt by countries. This will help to get a sense of the most common needs to get a loan across the countries.

In [7]:
%%sql
SELECT
    indicator_name,
    indicator_code,
    AVG(debt) AS average_debt
FROM 
    international_debt
GROUP BY 
    indicator_name, 
    indicator_code
ORDER BY 
    average_debt DESC

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
25 rows affected.


[('Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', 5904868401.499195),
 ('Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', 'DT.AMT.DPNG.CD', 5161194333.812659),
 ('Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', 2152041216.8902435),
 ('PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', 1958983452.8598351),
 ('PPG, private creditors (AMT, current US$)', 'DT.AMT.PRVT.CD', 1803694101.963265),
 ('Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', 1644024067.6508067),
 ('PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', 1223139290.39823),
 ('Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', 'DT.INT.DPNG.CD', 1220410844.4215188),
 ('PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', 1191187963.083064),
 ('PPG, bonds (AMT, current US$)', 'DT.AMT.PBND.CD', 1082623947.6536236),
 ('PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', 839843678.6158333),
 ('PPG, bonds (INT, current US$)', 'DT.INT.PBND.CD', 804733376.5985507),
 ('PPG, other private creditors (AMT, current US$)', 'DT.AMT.PROP.CD', 746888800.477778),
 ('PPG, commercial banks (AMT, current US$)', 'DT.AMT.PCBK.CD', 734868742.6011907),
 ('PPG, private creditors (INT, current US$)', 'DT.INT.PRVT.CD', 719740180.0887755),
 ('PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', 712619634.6598358),
 ('PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', 490062193.49838716),
 ('PPG, private creditors (DIS, current US$)', 'DT.DIS.PRVT.CD', 311323264.69056594),
 ('PPG, official creditors (INT, current US$)', 'DT.INT.OFFT.CD', 297677338.95725805),
 ('PPG, commercial banks (DIS, current US$)', 'DT.DIS.PCBK.CD', 293305195.5941176),
 ('PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', 164093285.6090164),
 ('PPG, commercial banks (INT, current US$)', 'DT.INT.PCBK.CD', 156647613.0559524),
 ('PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', 136230719.2451613),
 ('PPG, other private creditors (DIS, current US$)', 'DT.DIS.PROP.CD', 81135160.7),
 ('PPG, other private creditors (INT, current US$)', 'DT.INT.PROP.CD', 34250651.23518519)]

# 7. The highest amount of principal repayments
Principal repayments are used to get the immediate capital and pay it off in the long-term. This is the most popular type of debt indicator. Let's find out what country owes the most in this particular category. This can help us understand the specific state of the country and its specific economic needs.

In [8]:
%%sql
SELECT
    country_name,
    indicator_name,
    indicator_code,
    debt
FROM 
    international_debt
WHERE debt IN (SELECT MAX(debt)
              FROM international_debt
              WHERE indicator_code = 'DT.AMT.DLXF.CD')

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
1 rows affected.


[('China', 'Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', 96218620835.7)]

# 8. Top debt indicators
China owes 96 billion in debt repayments. On one hand, this amount sounds terrific, especially when it accounts only for 1 debt category. This number is equivalent to the current size of Ethiopian or Kenyan economies. On the other hand, this Chinese economy is the 2nd world largest and accounts for more than 14 bil USD according to [World Bank GDP Data](https://data.worldbank.org/indicator/NY.GDP.MKTP.CD?most_recent_value_desc=true).

In [9]:
%%sql
SELECT
    indicator_name,
    COUNT(indicator_code) AS indicator_count
FROM 
    international_debt
GROUP BY 
    indicator_name
ORDER BY 
    indicator_count DESC
FETCH FIRST 20 ROW ONLY

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
20 rows affected.


[('PPG, multilateral (INT, current US$)', 124),
 ('Principal repayments on external debt, long-term (AMT, current US$)', 124),
 ('Interest payments on external debt, long-term (INT, current US$)', 124),
 ('PPG, official creditors (INT, current US$)', 124),
 ('PPG, multilateral (AMT, current US$)', 124),
 ('PPG, official creditors (AMT, current US$)', 124),
 ('Disbursements on external debt, long-term (DIS, current US$)', 123),
 ('PPG, official creditors (DIS, current US$)', 122),
 ('PPG, bilateral (AMT, current US$)', 122),
 ('PPG, bilateral (INT, current US$)', 122),
 ('PPG, multilateral (DIS, current US$)', 120),
 ('PPG, bilateral (DIS, current US$)', 113),
 ('PPG, private creditors (AMT, current US$)', 98),
 ('PPG, private creditors (INT, current US$)', 98),
 ('PPG, commercial banks (INT, current US$)', 84),
 ('PPG, commercial banks (AMT, current US$)', 84),
 ('Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', 79),
 ('Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', 79),
 ('PPG, bonds (AMT, current US$)', 69),
 ('PPG, bonds (INT, current US$)', 69)]

# 9. The highest debt in each indicator by country & conclusion
There are 6 debt indicators in common for all the countries, hence there is a certain degree of mutuality when it comes to economic problems across different countries worldwide. 

The amount of debt the country requires for its specific needs depends on many factors, such as economy size, current necessities and development direction. Let's find out which countries owe the most across various debt indicators to get a sense of the economic issues specific countries to go through.

In this notebook, we dove deeper into the international debt data provided by the World Bank. We got an idea of what are the most typical economic issues around the globe, what is the total amount of debt and what are the most indebted countries.

In [10]:
%%sql
SELECT
    country_name,
    indicator_code,
    MAX(debt) AS maximum_debt
FROM
    international_debt
GROUP BY
    country_name,
    indicator_code
ORDER BY
    maximum_debt DESC
LIMIT 10

 * postgresql+psycopg2://postgres:***@localhost:5432/debt
10 rows affected.


[('China', 'DT.AMT.DLXF.CD', 96218620835.7),
 ('Brazil', 'DT.AMT.DLXF.CD', 90041840304.1),
 ('China', 'DT.AMT.DPNG.CD', 72392986213.8),
 ('Russian Federation', 'DT.AMT.DLXF.CD', 66589761833.5),
 ('Turkey', 'DT.AMT.DLXF.CD', 51555031005.8),
 ('South Asia', 'DT.AMT.DLXF.CD', 48756295898.2),
 ('Brazil', 'DT.AMT.PRVT.CD', 43598697498.6),
 ('Russian Federation', 'DT.AMT.DPNG.CD', 42800154974.9),
 ('Brazil', 'DT.AMT.DPNG.CD', 41831444053.3),
 ('Least developed countries: UN classification', 'DT.DIS.DLXF.CD', 40160766261.6)]